### ISSUE TITLE
transformer loss函数计算有误

### ISSUE THREAD

**Issue by ywbgithub** [candle-org/step_into_llm#61]
def forward(enc_inputs, dec_inputs):
    """前向网络
    enc_inputs: [batch_size, src_len]
    dec_inputs: [batch_size, trg_len]
    """
    logits, _, _, _ = model(enc_inputs, dec_inputs[:, :-1], src_pad_idx, trg_pad_idx)

    targets = dec_inputs[:, 1:].view(-1)
    loss = loss_fn(logits, targets)

    return loss

**Comment by ywbgithub** [comment]
logit数组 【batch_size, *src_len， trg_vocab_size】
targets数组【batch_size, *src_len】
两个数组维数不一致，而且对这两个数组使用loss_fn 没有实际意义

**Comment by ywbgithub** [comment]
logit数组 【batch_size, *src_len， trg_vocab_size】
targets数组【batch_size, *src_len】
两个数组维数不一致，而且对这两个数组使用loss_fn 没有实际意义
